app/src/main/java/net/sourceforge/opencamera/ExtendedAppInterface.java

```diff
@@ -209,27 +237,390 @@ public YuvImageUtils getYuvUtils() {
         return mYuvUtils;
     }
 
+    /**
+     * Starts RecSync by instantiating a {@link SoftwareSyncController}. If one is already running
+     * then it is closed and a new one is initialized.
+     * <p>
+     * Starts pick WiFI activity if neither WiFi nor hotspot is enabled.
+     */
     public void startSoftwareSync() {
         // Start softwaresync, close it first if it's already running.
-        if (mSoftwareSyncController != null) {
-            mSoftwareSyncController.close();
-            mSoftwareSyncController = null;
+        if (isSoftwareSyncRunning()) {
+            stopSoftwareSync();
         }
 
         try {
             mSoftwareSyncController =
                     new SoftwareSyncController(mMainActivity, null, new TextView(mMainActivity));
         } catch (IllegalStateException e) {
             // If wifi is disabled, start pick wifi activity.
-            Log.e(
-                    TAG,
-                    "Couldn't start SoftwareSync due to " + e + ", requesting user pick a wifi network.");
-            mMainActivity.finish(); // Close current app, expect user to restart.
-            mMainActivity.startActivity(new Intent(WifiManager.ACTION_PICK_WIFI_NETWORK));
+            requestWifiPick(TAG, "Couldn't start SoftwareSync due to " + e + ", requesting user to pick a wifi network.");
+            return;
+        }
+
+        // Listen for wifi or hotpot status changes.
```

Comment:
Please, explain what is happening below in terms of app behaviour -- why do we need to track states?

Reply:
It is in case a user switches Wi-Fi or hotspot off after RecSync has started. RecSync needs to be turned off, so I do the same as the original authors did: prompt the user to turn Wi-Fi back on.